improver_tests/calibration/test_init.py

```diff
@@ -103,94 +100,58 @@ def setUp(self):
         )
         realization_truth_1.attributes.update(truth_attributes)
         realization_truth_2.attributes.update(truth_attributes)
-        self.realization_truths = [realization_truth_1, realization_truth_2]
+        truths = [realization_truth_1, realization_truth_2]
 
-        self.landsea_mask = realization_truth_1.copy()
-        self.landsea_mask.rename("land_binary_mask")
+    additional_predictor = set_up_variable_cube(data=np.ones((4, 4), dtype=np.float32))
+    for coord in ["time", "forecast_reference_time", "forecast_period"]:
+        additional_predictor.remove_coord(coord)
+    additional_predictor.rename("kitten_present")
 
-    def test_probability_data(self):
-        """Test that when multiple probability forecast cubes and truth cubes
-        are provided, the groups are created as expected."""
+    landsea_mask = additional_predictor.copy(data=np.zeros((4, 4), dtype=np.float32))
+    landsea_mask.rename("land_binary_mask")
+    return forecasts, truths, [landsea_mask], [additional_predictor]
 
-        forecast, truth, land_sea_mask = split_forecasts_and_truth(
-            self.probability_forecasts + self.probability_truths, self.truth_attribute
-        )
 
-        self.assertIsInstance(forecast, iris.cube.Cube)
-        self.assertIsInstance(truth, iris.cube.Cube)
-        self.assertIsNone(land_sea_mask, None)
-        self.assertSequenceEqual((2, 2, 4, 4), forecast.shape)
-        self.assertSequenceEqual((2, 2, 4, 4), truth.shape)
-        self.assertTrue(np.all(forecast.data))
-        self.assertFalse(np.any(truth.data))
-
-    def test_probability_data_with_land_sea_mask(self):
-        """Test that when multiple probability forecast cubes, truth cubes, and
-        a single land-sea mask are provided, the groups are created as
-        expected."""
-
-        forecast, truth, land_sea_mask = split_forecasts_and_truth(
-            self.probability_forecasts + self.probability_truths + [self.landsea_mask],
-            self.truth_attribute,
-        )
+class Test_split_forecasts_and_truth(unittest.TestCase):
```

Comment:
I've had a go at condensing this with pytest and to avoid the `split_forecasts_and_truth` function using both unittest and pytest style unit tests, which I think would be preferable. We have used `pytest` parameterisation for this kind of exception testing e.g. in  [test_Combine.py](https://github.com/metoppv/improver/blob/master/improver_tests/cube_combiner/test_Combine.py#L105)

```
def cubes_for_exceptions() -> Tuple[CubeList, CubeList, Cube, str]:
    """Create input cubes for testing exceptions."""
    (realization_forecasts, realization_truths, land_sea_mask, _) = create_input_cubes(
        "realization"
    )
    truth_attribute = "mosg__model_configuration=uk_det"
    return realization_forecasts, realization_truths, land_sea_mask, truth_attribute


@pytest.mark.parametrize(
    "condition, error_class, msg",
    (
        ("multiple_landmasks", IOError, "Expected at most one cube for land-sea mask."),
        (
            "name_mismatch",
            ValueError,
            "Only forecasts for one diagnostic can be input.",
        ),
        ("missing_truths", IOError, "Missing truth input."),
        ("missing_forecasts", IOError, "Missing historical forecast input."),
    ),
)
def test_exceptions(condition, error_class, msg):
    """Test that when multiple land-sea masks are provided an exception is
    raised."""
    (
        realization_forecasts,
        realization_truths,
        land_sea_mask,
        truth_attribute,
    ) = cubes_for_exceptions()

    land_sea_mask_name = land_sea_mask[0].name()
    if condition == "multiple_landmasks":
        land_sea_mask = land_sea_mask + land_sea_mask
    elif condition == "name_mismatch":
        realization_truths[0].rename("kitten_density")
    elif condition == "missing_truths":
        realization_truths = []
    elif condition == "missing_forecasts":
        realization_forecasts = []

    with pytest.raises(error_class, match=msg):
        split_forecasts_and_truth(
            realization_forecasts + realization_truths + land_sea_mask,
            truth_attribute,
            land_sea_mask_name
        )
```

Reply:
Done - thank you for the code implementation suggestion, I've mostly kept this test as above but changed the test name